module/forms/level-up-form.js

```diff
@@ -102,7 +102,18 @@ export class PTULevelUpForm extends FormApplication {
     const itemIdsToDelete = [];
     const itemsToAdd = [];
     const state = {...this.store.state};
-    const lostHealth = this.object.actor.system.health.max - this.object.actor.system.health.value;
+    const oldData = {
+      system: {
+        level:{
+          "current": state.changeDetails.oldLvl,
+          "exp": state.changeDetails.oldExp
+        }
+      }
+    }
+    await this.object.actor.update(oldData);
```

Comment:
Why are you updating the actor here to something it already is?

Also `current` should never be directly set, it's a derrived value.

Reply:
I just read your comment below; yeah let's not do this, way too many database calls for a very simple function that realistically doesn't *do* much in the grand scheme of this.

Either manually calculate the new health value, or don't bother

Reply:
manually calculated old health values to get missing health, then manually calculated the new max health then subtracted the missing health to get the new current health

```js
const oldHealthTotal = 10 + state.changeDetails.oldLvl + (this.object.actor.system.stats.hp.total * 3);
const oldHealthMax = this.object.actor.system.health.injuries > 0 ? Math.trunc(oldHealthTotal * (1 - ((this.object.actor.system.modifiers.hardened ? Math.min(this.object.actor.system.health.injuries, 5) : this.object.actor.system.health.injuries) / 10))) : oldHealthTotal;
const oldHealthValue = this.object.actor.system.health.value ?? oldHealthMax;

const missinghealth = oldHealthMax - oldHealthValue;
```
   


```js
//adjust health as per level up
const newHealthTotal = 10 + state.changeDetails.newLvl + (this.object.actor.system.stats.hp.total * 3);
const newHealthMax = this.object.actor.system.health.injuries > 0 ? Math.trunc(newHealthTotal * (1 - ((this.object.actor.system.modifiers.hardened ? Math.min(this.object.actor.system.health.injuries, 5) : this.object.actor.system.health.injuries) / 10))) : newHealthTotal;
const newhealth = newHealthMax - missinghealth;
```